pkg/controllers/deprovisioning/controller.go

```diff
@@ -107,19 +115,13 @@ func (c *Controller) LivenessProbe(_ *http.Request) error {
 }
 
 func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
-	// the last cluster consolidation wasn't able to improve things and nothing has changed regarding
-	// the cluster that makes us think we would be successful now
-	if c.lastConsolidationState == c.cluster.ClusterConsolidationState() {
-		return reconcile.Result{RequeueAfter: pollingPeriod}, nil
-	}
-
 	result, err := c.ProcessCluster(ctx)
 	if err != nil {
 		return reconcile.Result{}, fmt.Errorf("deprovisioning cluster, %w", err)
 	} else if result == ResultRetry {
 		return reconcile.Result{Requeue: true}, nil
 	} else if result == ResultNothingToDo {
-		c.lastConsolidationState = c.cluster.ClusterConsolidationState()
+		c.consolidation.lastConsolidationState = c.cluster.ClusterConsolidationState()
```

Comment:
This leaks consolidation details that are encapsulated in the Consolidation deprovisioner out into the standard provisioner, which feels wrong

Reply:
I can see why you think that. This is the only deprovisioner that might save some computation by being aware of the previous deprovisioning actions. I'll try and tackle this in the further simplification. Sound good?